### PR TITLE
Issue-5887 Option for recursive check if node is enabled

### LINKED ISSUE #5887: gui.is_enabled() parameter to check if some nodes in hierarchy disabled.
**Is your feature request related to a problem? Please describe (REQUIRED):**
When we do `gui.pick_node()`on button we need to check if this button enabled using `gui.is_enabled()`.

In complicated GUI it's possible that some parent node is off, in this case we have to check whole hierarhy of nodes in loop which is not so efficient. It's a pretty frequent case and it would be nice if it will be covered out of the box.

**Describe the solution you'd like (REQUIRED):**
Although a node with disabled parent behaves the same way as disabled node, I'm still not sure we can change `gui.is_enabled()` behaviour without back compatability.

I think it should be a separate argument for  `gui.is_enabled()` function, like this:

`local is_enabled = gui.is_enabled(node)` - check if the node itself disabled, as it's now

`local is_enabled = gui.is_enabled(node, true)` - check if node itself or any parent node is disabled.

**Describe alternatives you've considered (REQUIRED):**
Iterate all the parents in Lua:
https://github.com/Insality/druid/blob/63a6af3faa643d13119c5f35085b601c4543d4e0/druid/helper.lua#L160

### PR DESCRIPTION
**Pull request by AGulev**
Fix https://github.com/defold/defold/issues/5887

### CODE REVIEW

**Comment by AGulev** [comment]
@britzl  I created it as draft to think if we wanna have this functionality.

File: engine/gui/src/gui_script.cpp [hunk 1]
```diff
@@ -2841,17 +2841,21 @@ namespace dmGui
...
      */
     static int LuaIsEnabled(lua_State* L)
     {
+        int top = lua_gettop(L);
         HNode hnode;
         InternalNode* n = LuaCheckNode(L, 1, &hnode);
         (void) n;
 
         Scene* scene = GuiScriptInstance_Check(L);
+        bool recursive = top == 2 && lua_toboolean(L, 2);
```
Comment (britzl) [inline engine/gui/src/gui_script.cpp:2855]: What if I've accidentally called the function with three arguments?

`gui.pick_node(node, true, "foobar")`

Perhaps better to check `top >= 2`?

File: engine/gui/src/gui_script.cpp [hunk 2]
```diff
@@ -2841,17 +2841,21 @@ namespace dmGui
...
+        lua_pushboolean(L, dmGui::IsNodeEnabled(scene, hnode, recursive));
 
-        lua_pushboolean(L, dmGui::IsNodeEnabled(scene, hnode));
+        assert(top + 1 == lua_gettop(L));
```
Comment (britzl) [inline engine/gui/src/gui_script.cpp:2858]: I think we should change to DM_LUA_STACK_CHECK every time we update a function like this one.

**Review by britzl — CHANGES_REQUESTED** [review]
Great initiative! Please check my comment about the stack check

File: engine/gui/src/gui_script.cpp (hunk 1 above)
Comment (AGulev) [reply engine/gui/src/gui_script.cpp]: Thank you, totally makes sense.

File: engine/gui/src/gui_script.cpp (hunk 2 above)
Comment (AGulev) [reply engine/gui/src/gui_script.cpp]: Thx! Fixed!

**Review by britzl — APPROVED** [review]
Looks good!

**Pull request merged**